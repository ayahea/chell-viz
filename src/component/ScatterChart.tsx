import * as React from 'react';

import { RESIDUE_TYPE } from '../data/chell-data';

import PlotlyChart, { defaultConfig, defaultLayout, generateScatterGLData } from '../helper/PlotlyHelper';
import { withDefaultProps } from '../helper/ReactHelper';

export const defaultScatterChartProps = {
  candidateResidues: new Array<RESIDUE_TYPE>(),
  height: 400,
  hoveredResidues: new Array<RESIDUE_TYPE>(),
  onClickCallback: (...args: any[]) => {
    return;
  },
  onHoverCallback: (...args: any[]) => {
    return;
  },
  onSelectedCallback: (...args: any[]) => {
    return;
  },
  onUnHoverCallback: (...args: any[]) => {
    return;
  },
  width: 400,
};

export type ScatterChartProps = {
  data: Array<{
    color: string;
    points: Array<{ i: number; j: number }>;
  }>;
  nodeSize: number;
} & typeof defaultScatterChartProps;

class ScatterChartClass extends React.Component<ScatterChartProps, any> {
  constructor(props: any) {
    super(props);
  }
  public render() {
    const { candidateResidues, height, hoveredResidues, data, nodeSize, width, ...props } = this.props;

    const scatterGLData = data.map(entry => generateScatterGLData(entry.points, entry.color, nodeSize, true));

    return (
      <PlotlyChart
        config={{
          ...defaultConfig,
        }}
        data={scatterGLData}
        layout={{
          ...defaultLayout,
          height,
          width,
          xaxis: {
            ...defaultLayout.xaxis,
            gridcolor: '#ff0000',
            gridwidth: nodeSize,
            showticklabels: false,
            tickvals: [...candidateResidues, ...hoveredResidues],
          },
          yaxis: {
            ...defaultLayout.yaxis,
            gridcolor: '#ff0000',
            gridwidth: nodeSize,
            showticklabels: false,
            tickvals: [...candidateResidues, ...hoveredResidues],
          },
        }}
        {...props}
      />
    );
  }
}

export const ScatterChart = withDefaultProps(defaultScatterChartProps, ScatterChartClass);

export default ScatterChart;
