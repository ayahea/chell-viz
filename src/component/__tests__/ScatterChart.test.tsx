import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';

import ScatterChart from '../ScatterChart';

describe('ScatterChart', () => {
  const emptyData = [
    {
      color: '',
      points: [],
    },
  ];

  test('Should match existing snapshot when given simple data.', () => {
    const wrapper = shallow(<ScatterChart data={emptyData} nodeSize={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
