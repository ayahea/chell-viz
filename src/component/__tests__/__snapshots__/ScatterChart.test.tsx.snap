// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScatterChart Should match existing snapshot when given simple data. 1`] = `
<PlotlyChart
  config={
    Object {
      "displayModeBar": true,
    }
  }
  data={
    Array [
      Object {
        "marker": Object {
          "color": "",
          "size": 8,
        },
        "mode": "markers",
        "type": "scattergl",
        "x": Array [],
        "y": Array [],
      },
    ]
  }
  layout={
    Object {
      "height": 400,
      "legend": Object {},
      "showlegend": false,
      "title": "",
      "width": 400,
      "xaxis": Object {
        "gridcolor": "#ff0000",
        "gridwidth": 4,
        "showticklabels": false,
        "tickvals": Array [],
      },
      "yaxis": Object {
        "autorange": "reversed",
        "gridcolor": "#ff0000",
        "gridwidth": 4,
        "showticklabels": false,
        "tickvals": Array [],
      },
    }
  }
  onClickCallback={[Function]}
  onHoverCallback={[Function]}
  onSelectedCallback={[Function]}
  onUnHoverCallback={[Function]}
/>
`;
